### PR TITLE
Windows Event Log Support

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Adds support for the Windows Event Log. This is a port from the develop branch.

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/pull/54713

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

File: tests/pytests/unit/modules/test_win_event.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+
+def test_count():
+    """
+    Test win_event.count
+    """
+    ret = win_event.count("System")
+    assert ret > 0
+
+
```
Comment (damon-atkins) [inline tests/pytests/unit/modules/test_win_event.py:55]: A comment,   like search/get a unknown Windows Event Log to confirm this functions.

**Review by damon-atkins — CHANGES_REQUESTED** [review]
Please add more comments to the code........     One Typo....   👍

File: salt/modules/win_event.py [hunk 2]
```diff
@@ -0,0 +1,754 @@
...
+    Returns:
+        PyHANDLE: A handle to the event log
+    """
+
+    # TODO: upgrade windows token
+    # "log close" can fail if this is not done
+    try:
+        return win32evtlog.OpenEventLog(None, log_name)
+    except pywintypes.error:
+        raise FileNotFoundError(
```
Comment (s0undt3ch) [inline salt/modules/win_event.py:200]: The actual exception message/data is not useful?

File: tests/pytests/unit/modules/test_win_event.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
+import datetime
+
+import pytest
+
+import salt.modules.win_event as win_event
+
+pytestmark = [
+    pytest.mark.windows_whitelisted,
+    pytest.mark.skip_unless_on_windows,
+    pytest.mark.destructive_test,
```
Comment (s0undt3ch) [inline tests/pytests/unit/modules/test_win_event.py:10]: Why is it destructive?

Comment (twangboy) [reply tests/pytests/unit/modules/test_win_event.py:10]: It clears the application log, then sets creates some logs in that application log, and then clears it at the end

File: tests/pytests/unit/modules/test_win_event.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+]
+
+
+@pytest.fixture(scope="function")
+def application_events():
+    win_event.clear("Application")
```
Comment (damon-atkins) [inline tests/pytests/unit/modules/test_win_event.py:16]: Please add a comment this delete the contents of the event log.  Does this need to be run with Admin access?

File: salt/modules/win_event.py (hunk 2 above)
Comment (twangboy) [reply salt/modules/win_event.py:200]: Now using exe.stderror

File: tests/pytests/unit/modules/test_win_event.py (hunk 4 above)
Comment (twangboy) [reply tests/pytests/unit/modules/test_win_event.py:16]: Done

File: salt/modules/win_event.py [hunk 5]
```diff
@@ -0,0 +1,754 @@
...
+
+    Args:
+
+        time (ob): A datetime object
+
+    Returns:
+        TimeTuple: A TimeTuple
+    """
+
+    return TimeTuple._make(
```
[the remark this answers is not in the document]
Comment (twangboy) [reply salt/modules/win_event.py]: Done

File: tests/pytests/unit/modules/test_win_event.py (hunk 3 above)
Comment (twangboy) [reply tests/pytests/unit/modules/test_win_event.py]: Marked the specific destructive tests

File: salt/modules/win_event.py [hunk 6]
```diff
@@ -0,0 +1,754 @@
...
+    "reserved",
+    "reservedFlags",
+    "sid",
+    "sourceName",
+    "stringInserts",
+    "timeGenerated",
+    "timeWritten",
+)
+
+EVENT_TYPES = {
```
Comment (Ch3LL) [inline salt/modules/win_event.py:43]: We should avoid using mutable objects at the global scope. It looks like this is only being used in one function, can we move this and the other variables here to the functions?

File: salt/modules/win_event.py [hunk 7]
```diff
@@ -0,0 +1,754 @@
+"""
```
Comment (Ch3LL) [inline salt/modules/win_event.py:1]: need a versionadded here for 3006

**Review by Ch3LL — CHANGES_REQUESTED** [review]

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**